<a href="https://colab.research.google.com/github/Irina-Bichik/-Final-examination/blob/main/%D0%98%D1%82%D0%BE%D0%B3%D0%BE%D0%B2%D0%B0%D1%8F_%D0%B0%D1%82%D1%82%D0%B5%D1%81%D1%82%D0%B0%D1%86%D0%B8%D1%8F_%D0%97%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import math
import scipy.stats as stats

Мы провели АБ-тест на увеличение average timespent  per user. По итогам тестирования мы получили следующие данные. Является ли результат статистически значимым с уровнем доверия 80%? Какую версию мы выкатим на продакшн?

A) Средняя - 360, отклонение - 40, количество - 9802

B) Средняя - 352, отклонение - 58, количество – 9789


Н0 - средние значения времени пользователей на версии А и версии В равны

Н1 - средние значения времени пользователей на версии А и версии В различаются

Используем t-критерий Стьюдента

Вычислим стандартную ошибку среднего для версий А и В

In [8]:
X_A = 360
σ_A = 40
n_A = 9802
X_B = 352
σ_B = 58
n_B = 9789

In [7]:
SE_A = σ_A/math.sqrt (n_A)
SE_B = σ_B/math.sqrt (n_B)

print (f'SE версия A: {round(SE_A,3)}')
print (f'SE версия B: {round(SE_B,3)}')


SE версия A: 0.404
SE версия B: 0.586


Вычислим t-статистику

In [13]:
t = (X_A-X_B)/(math.sqrt(SE_A**2+SE_B**2))
print (f' Расчетное значение t = {t: .3f}')

 Расчетное значение t =  11.237


In [23]:
df = n_A+n_B-2
print (f' Степень свободы df = {df: .0f}')

 Степень свободы df =  19589


Для уровня доверия 80% и степеней свободы 19589 критическое значение t- статистики по таблице равно 1,282

Так как расчетное значение t-статистики больше критического значения (11,237 > 1.282), то принимается альтернативная гипотеза Н1 о том, что средние значения времени, проведенного пользователями на версиях А и В различаются. Результат является статистически значимым с уровнем доверия 80%. Следовательно мы выкатим в продакшн версию А.